src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/IdentifiersShouldNotContainUnderscoresTests.cs

```diff
@@ -439,6 +440,73 @@ public Span(string text)
 ");
         }
 
+        [Fact]
+        public async Task CA1707_CSharp_DiscardSymbolParameter_NoDiagnostic()
+        {
+            await VerifyCS.VerifyAnalyzerAsync(@"
+public static class MyHelper
+{
+    public static int GetBar(this string _) => 42;
```

Comment:
💭 I'm not sure this is a good thing. If this was my code, I would still expect a proper parameter name here. Tagging @mavasani for discussion.

Reply:
I see your point. Maybe we could allow discard parameters only for interface implementation or overrides, WDYT?

Reply:
ping @sharwell

Reply:
Proper parameter name here would generate CA1801 (https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1801?view=vs-2019)

```
If you're using FxCop analyzers, this rule does not flag parameters that are named with the discard symbol, for example, _, _1, and _2. This reduces warning noise on parameters that are needed for signature requirements, for example, a method used as a delegate, a parameter with special attributes, or a parameter whose value is implicitly accessed at run time by a framework but is not referenced in code.
```